tests/GuardTest.php

```diff
@@ -318,6 +318,43 @@ public function testTokenIsRemovedFromStorageWhenPersistentModeIsOff()
         self::assertArrayNotHasKey('test_name', $storage);
     }
 
+    public function testTokenInBodyofGetIsInvalid()
+    {
+        $storage = [
+            'test_name' => 'test_value123',
+        ];
+
+        // we set up a failure handler that we expect to be called because a GET cannot have a token
+        $self = $this;
+        $failureHandlerCalled = 0;
+        $failureHandler = function () use ($self, &$failureHandlerCalled) {
+            $failureHandlerCalled++;
+            $responseProphecy = $self->prophesize(ResponseInterface::class);
+            return $responseProphecy->reveal();
+        };
+
+        $responseFactoryProphecy = $this->prophesize(ResponseFactoryInterface::class);
+
+        $mw = new Guard($responseFactoryProphecy->reveal(), 'test', $storage, $failureHandler);
+
+        $requestHandlerProphecy = $this->prophesize(RequestHandlerInterface::class);
+
+        $requestProphecy = $this->prophesize(ServerRequestInterface::class);
+        $requestProphecy
+            ->getMethod()
+            ->willReturn('GET')
+            ->shouldBeCalledOnce();
+        $requestProphecy
+            ->getParsedBody()
+            ->willReturn([
+                'test_name' => 'test_name',
+                'test_value' => 'test_value123',
+            ])
+            ->shouldBeCalledOnce();
```

Comment:
This is a weird test case, you cannot pass a body with a `GET` request, the only way to pass parameters is via query params which would never end up being parsed unless you accessed them via `getQueryParams()`

Reply:
It's actually completely possible:

Given this:
```
$app->get('/', function ($request, $response){
    $posted = $request->getParsedBody();
    $response->getBody()->write(json_encode(['posted data' => $posted]));
    return $response;
});
```

Then this will happen:

```
$ curl -X GET -d "foo=bar" localhost:8888
{"posted data":{"foo":"bar"}}
```

Reply:
That’s such an oddity. Good to know!